### DISCUSSION TITLE
setting a titleTemplate; setting a title from page

### DISCUSSION THREAD

**Discussion by nachtfunke** [nuxt/nuxt#18143] [Questions] [answered]
I'm trying to change the title of my website. But I cannot figure out how. I either get breaking errors or nothing happens at all. I am not sure, maybe this feature is just not implemented in nuxt 3 yet?

If so, I was wondering if anyone could point me to the documentation on this. I cannot find it and whatever is documented in the old docs does not work / throws errors.

**Comment by wobsoriano** [comment, marked as answer]
```ts
useMeta({ title: 'My page' })
```

https://v3.nuxtjs.org/docs/usage/meta-tags

**Reply by nachtfunke** [reply]
Thank you. I used that from the beginning, but it didn't work. I realise now that this was because I was trying to add from data into the title which apparently does not work.

**Reply by nachtfunke** [reply]
I even found [this article](https://langvad.dev/blog/how-to-work-with-meta-data-in-nuxt/), specifically saying that setting a `titleTemplate` should be possible in `nuxt.config.ts`, yet when I do that, my application crashes.

Even though this article claims that this is possible, I find nothing of this option on the nuxt documentation: https://v3.nuxtjs.org/docs/directory-structure/nuxt.config

It speaks of `meta` and for some reason considers `<link>` & `<script>` to be specifically part of it, but `<title>` gets left out entirely. The nuxt 2 doc lets me believe this _should_ be possible. Was this feature removed from nuxt 3 and is this the reason why it isn't in the docs anymore? Or maybe I am just too incompetent to navigate the docs.

**Reply by wobsoriano** [reply]
Do you have a simple repro of how are you doing it? thanks

**Reply by nachtfunke** [reply]
I can't share my repo at the moment unfortunately :/ Maybe this helps:

In my `nuxt.config.ts`:
```javascript
export default defineNuxtConfig({
    meta: {
        head: {
            titleTemplate: '%s - website',
        },
    }
})
```
I originally read it as just the `head`, not nested in `meta`, but that crashed my application. This way, the application doesn't crash, but it also doesn't work.

In any of the pages in `/pages`:
```javascript
<script>
    import { useMeta } from "nuxt3/dist/meta/runtime"
    
    export default {
        setup() {
            useMeta({
                title: this.title
            })
        },
        data() {
            return {
                title: "Termin vereinbaren"
            }
        },
    }
</script>
```
This doesn't work. The title will not be passed, instead no title will be set at all. If I replace the `this.title` instead with some string, it will set _only_ that string.

I am trying to make this all work with `titleTemplate` _and_ passing a string from `data()` into this title template from a page.

**Reply by viii47** [reply]
@nachtfunke This is because you do not have access to `data()`  from the `setup()` *to my knowledge*, however there's a lemon-squeezy solution:

```js

<template>
  <div>
    <Html>
      <Head>
        <Title>{{ title }}</Title>
      </Head>
    </Html>
    <h1>My Page</h1>
    [... your super dope template code here]
  </div>
</template>
```

```js
  data() {
    return {
      title: 'testmeeeeeeeeeee'
    }
  },
```

**Note:** The meta tags `<Html>, <Title>, ...` have to be capitalized so nuxt knows to inject them properly.

**Reply by nachtfunke** [reply]
Right, setting the title works now! But I also need to be able to set `titleTemplate`. I don't wanna have to repeat the same string all the time. And implementing a store pattern just for carrying through a simple string seems overkill. I joined the discord and after some pestering I actually got the answer that `titleTemplate` just isn't supported anymore in nuxt 3. I was prompted to make a [Feature Request](https://github.com/nuxt/nuxt.js/issues/10073) instead
